# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsDOMWindowUtils.cpp
# Commit: 48eaa84c206d
# Full Hash: 48eaa84c206dbdda637c57b55ddbd07e975ba103
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2020-06-23 16:15:06
# Regressor Bug: 1637500
# File Overlap Count: 2
# Description:
#   Bug 1637500 - P2. Rename methods as they are not always dealing with "threads". r=froydnj
#   
#   Before P1, GetCurrentThreadSerialEventTarget would have always returned the same data as NS_GetCurrentThread, making the comment incorrect Now it will properly return the running TaskQueue if any.
#   
#   This change of name more clearly exposes what they are doing, as we aren't always dealing with threads directly; but a nsISerialEventTarget
# ==============================================================================

diff -r 00687d06a8a6 -r 48eaa84c206d dom/base/nsDOMWindowUtils.cpp
--- a/dom/base/nsDOMWindowUtils.cpp	Tue Jun 23 05:02:39 2020 +0000
+++ b/dom/base/nsDOMWindowUtils.cpp	Tue Jun 23 05:05:36 2020 +0000
@@ -4206,7 +4206,7 @@
   } else {
     cbc->SendBeginRecording(TimeStamp::Now())
         ->Then(
-            GetCurrentThreadSerialEventTarget(), __func__,
+            GetCurrentSerialEventTarget(), __func__,
             [promise](const bool& aSuccess) {
               if (aSuccess) {
                 promise->MaybeResolve(true);
@@ -4247,7 +4247,7 @@
     promise->MaybeReject(NS_ERROR_UNEXPECTED);
   } else if (aWriteToDisk) {
     cbc->SendEndRecordingToDisk()->Then(
-        GetCurrentThreadSerialEventTarget(), __func__,
+        GetCurrentSerialEventTarget(), __func__,
         [promise](const bool& aSuccess) {
           if (aSuccess) {
             promise->MaybeResolveWithUndefined();
@@ -4262,7 +4262,7 @@
         });
   } else {
     cbc->SendEndRecordingToMemory()->Then(
-        GetCurrentThreadSerialEventTarget(), __func__,
+        GetCurrentSerialEventTarget(), __func__,
         [promise](Maybe<CollectedFramesParams>&& aFrames) {
           if (!aFrames) {
             promise->MaybeRejectWithUnknownError(